src/systemctl/systemctl.c

```diff
@@ -6326,6 +6326,119 @@ static int log_level(int argc, char *argv[], void *userdata) {
         return 0;
 }
 
+static int service_name_to_dbus(sd_bus *bus, const char *name, char **ret_dbus_name) {
+        _cleanup_(sd_bus_error_free) sd_bus_error error = SD_BUS_ERROR_NULL;
+        _cleanup_strv_free_ char **names = NULL;
+        const char *p, *q;
+        int r;
+
+        p = startswith(name, "dbus-");
+
+        if (!p) {
+                _cleanup_free_ char *dbus_path = NULL;
+
+                dbus_path = unit_dbus_path_from_name(name);
+                if (!dbus_path)
+                        return log_oom();
+
+                r = sd_bus_get_property_strv(
+                                bus,
+                                "org.freedesktop.systemd1",
+                                dbus_path,
+                                "org.freedesktop.systemd1.Unit",
+                                "Names",
+                                &error,
+                                &names);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to determine Names property of %s: %s",
+                                               name, bus_error_message(&error, r));
+
+                char **s;
+                STRV_FOREACH(s, names) {
+                        p = startswith(*s, "dbus-");
+                        if (p)
+                                break;
+                }
+        }
+
+        if (!p)
+                return log_error_errno(SYNTHETIC_ERRNO(ENOLINK),
+                                       "Unit %s doesn't declare any name with \"dbus-\" prefix.", name);
+
+        q = endswith(p, ".service");
+        if (!q)
+                return log_error("%s is not a service, cannot set log level for it.", p);
```

Comment:
This should return negative errno. Also, `cannot set log level for it.` should be dropped or updated, as this is also used for `systemctl service-log-target`.